app/Jobs/CirrusSearch/QueueSearchIndexBatches.php

```diff
@@ -86,12 +97,13 @@ public function handleResponse( string $rawResponse, $error ) : void {
             foreach ($batches as $job) {
                 $this->dispatch($job);
             }
-
         } else {
-
             Log::error(__METHOD__ . ": Job finished but didn't create any batches, something is weird");
-            $this->fail( new \RuntimeException($this->apiModule() . ' call for '.$this->wikiId.' was not successful:' . $rawResponse ) );
+            $this->fail( new \RuntimeException($this->apiModule() . ' call for ' . $this->wikiId() . ' was not successful:' . $rawResponse ) );
         }
+    }
 
+    protected function getQueryParams(): string {
+        return parent::getQueryParams() . '&cluster=' . $this->cluster();
```

Comment:
@AndrewKostka I'm currently looking into moving this forward and there's a high-level question I'd like to ask here: what's the reason you decided to have api be in control over which clusters the maintenance scripts are invoked for, when instead this could also be handled by our wrappers in MediaWiki themselves? E.g. by default it seems most CirrusSearch maintenance scripts already work on all known clusters by default.

Reply:
> AFAIK the CirrusSearch maintenance scripts do not work on all clusters by default.

Interesting. The script level documentation / comments say differently, but it's probably best to be explicit in that case.